apps/api/src/chat/services/text-processing.service.ts

```
@@ -0,0 +1,83 @@
+      .join(' ')
+      .slice(0, 4000); // Reasonable limit to prevent memory issues
+  }
+
+  sanitizeText(text: string): string {
+    if (typeof text !== 'string') {
+      return '';
+    }
+    
+    return text
+      .trim()
+      .replace(/[\u0000-\u001F\u007F-\u009F]/g, '') // Remove control characters
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)